### PR TITLE
libsql: provide more return info for sync

### PR DESCRIPTION
**Pull request by LucioFranco**
This PR changes the return type of sync to a struct that provides the current commit fno and how many frames that sync call required. This should give more insight to users to understand how much data has been synced.

### CODE REVIEW

File: libsql-server/tests/embedded_replica/mod.rs [hunk 1]
```diff
@@ -1226,3 +1226,134 @@ fn txn_bug_issue_1283() {
...
+        assert_eq!(rep.start_frame_no(), Some(1));
+
+        notify.notify_waiters();
+        notify.notified().await;
+
+        tokio::time::sleep(std::time::Duration::from_secs(5)).await;
+
+        let rep = db.sync().await.unwrap();
+        assert_eq!(rep.frame_no(), Some(4));
+        assert_eq!(rep.start_frame_no(), Some(4));
```
Comment (LucioFranco) [inline libsql-server/tests/embedded_replica/mod.rs:1353]: I would expect this to be 0 or 1 but it doesn't seem to re replicate, any ideas @MarinPostma

File: libsql/src/replication/mod.rs [hunk 2]
```diff
@@ -229,7 +255,20 @@ impl EmbeddedReplicator {
             }
         }
 
-        Ok(replicator.client_mut().committed_frame_no())
+        let last_frames_synced = self.last_frames_synced.fetch_add(
+            replicator.frames_synced(),
+            std::sync::atomic::Ordering::Relaxed,
```
Comment (haaawk) [inline libsql/src/replication/mod.rs:260]: I usually recommend using sequential ordering unless you know this is a bottleneck in any way.

**Review by haaawk — APPROVED** [review]
It is not helping in review that you're reformatting code not changed in the PR.

File: libsql/src/replication/mod.rs (hunk 2 above)
Comment (LucioFranco) [reply libsql/src/replication/mod.rs:260]: While I don't disagree, I think the fact that this isn't being ordered with anything else + being a fetch_add means that these orderings don't mean much in the end and thus relaxed is fine.

**Comment by LucioFranco** [comment]
> It is not helping in review that you're reformatting code not changed in the PR.

I am not sure why my editor is formatting something but running `cargo fmt` isn't. I need to look into that since those changes should be enforced by CI.

**Pull request merged**